src/components/Provider/BreakpointProvider.tsx

```diff
@@ -1,44 +1,35 @@
 import debounce from 'lodash/debounce';
 import * as React from 'react';
-import {
-	Breakpoint,
-	BreakpointContext,
-} from '../../contexts/BreakpointContext';
+import { BreakpointContext } from '../../contexts/BreakpointContext';
 export interface BreakpointProviderProps {
 	breakpoints: number[];
-	onBreakpointChange?: (breakpoint: Breakpoint) => void;
+	onBreakpointChange?: (breakpoint: number) => void;
 }
 
 export const BreakpointProvider: React.FC<BreakpointProviderProps> = ({
 	breakpoints,
 	children,
 }) => {
-	const [currentBreakpoint, setCurrentBreakpoint] = React.useState(
-		0 as Breakpoint,
-	);
-
-	const handleResize = () => {
-		const clientWidth = window.innerWidth;
-		if (clientWidth < breakpoints[0]) {
-			setCurrentBreakpoint(0);
-		} else if (clientWidth < breakpoints[1]) {
-			setCurrentBreakpoint(1);
-		} else if (clientWidth < breakpoints[2]) {
-			setCurrentBreakpoint(2);
-		} else if (clientWidth >= breakpoints[2]) {
-			setCurrentBreakpoint(3);
-		}
-	};
+	const [currentBreakpoint, setCurrentBreakpoint] = React.useState<number>(0);
 
 	React.useEffect(() => {
-		const debouncedResize = debounce(handleResize, 80);
+		if (breakpoints && !!breakpoints.length) {
```

Comment:
Let's reduce nesting by checking the opposite condition and returning early (`if (!breakpoints..) return`)

Reply:
This doesn't seem to have been resolved, let's return early here.